packages/xo-web/src/xo-app/hub/resource.js

```diff
@@ -0,0 +1,260 @@
+import _ from 'intl'
+import ActionButton from 'action-button'
+import decorate from 'apply-decorators'
+import Icon from 'icon'
+import React from 'react'
+import { Card, CardBlock, CardHeader } from 'card'
+import { Col, Row } from 'grid'
+import { alert, form } from 'modal'
+import { connectStore, formatSize, getXoaPlan } from 'utils'
+import { createGetObjectsOfType } from 'selectors'
+import { downloadAndInstallResource, deleteTemplates } from 'xo'
+import { error, success } from 'notification'
+import { find, filter } from 'lodash'
+import { injectState, provideState } from 'reaclette'
+import { withRouter } from 'react-router'
+
+import ResourceForm from './resource-form'
+
+const subscribeAlert = () =>
+  alert(
+    _('hubResourceAlert'),
+    <div>
+      <p>
+        {_('considerSubscribe', {
+          link: 'https://xen-orchestra.com',
+        })}
+      </p>
+    </div>
+  )
+
+export default decorate([
+  withRouter,
+  connectStore(() => {
+    const getTemplates = createGetObjectsOfType('VM-template').sort()
+    const getPools = createGetObjectsOfType('pool')
+    return {
+      templates: getTemplates,
+      pools: getPools,
+      hubInstallLoadingState: state => state.hubInstallLoadingState,
+    }
+  }),
+  provideState({
+    initialState: () => ({
+      selectedInstallPools: [],
+    }),
+    effects: {
+      async install() {
+        const {
+          id,
+          name,
+          namespace,
+          setHubInstallLoadingState,
+          version,
+        } = this.props
+        const { hubInstallLoadingState, isTemplateInstalled } = this.state
+        if (getXoaPlan(process.env.XOA_PLAN) === 'Community') {
+          subscribeAlert()
+          return
+        }
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm
+              install
+              multi
+              poolPredicate={isTemplateInstalled}
+              {...props}
+            />
+          ),
+          header: (
+            <span>
+              <Icon icon='add-vm' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+
+        setHubInstallLoadingState({
+          ...hubInstallLoadingState,
+          [id]: true,
+        })
```

Comment:
This should be done in the reducer, the actions should be as simple to use as possible.